Makefiles/sledconf.esp32

```diff
@@ -0,0 +1,59 @@
+PROJECT := sled
+DEBUG := 0
+STATIC := 1
+PLATFORM := freertos
+DEFAULT_OUTMOD := esp32
+DEFAULT_MODULEDIR := "./modules"
+
+## required output module
+MODULES := out_$(DEFAULT_OUTMOD)
+
+## single modules for testing purposes
+MODULES += gfx_twinkle
+MODULES += gfx_partirush
+MODULES += gfx_golc
+MODULES += gfx_matrix
+MODULES += gfx_rainbow
+MODULES += gfx_balls
+MODULES += gfx_random_rects
+MODULES += gfx_reddot
+MODULES += gfx_plasma
+MODULES += gfx_sparkburn
+MODULES += gfx_ursuppe2
+MODULES += gfx_xorrid
+
+## Build all background and filter modules
+#MODULES += $(BGMMODS_DEFAULT)
+MODULES += $(FLTMODS_DEFAULT)
+
+## Build all gfx modules (from GNUmakefile)
+## Comment this out if you want to build only single modules
+## Note: Some are not working yet on esp32
+#MODULES += $(GFXMODS_DEFAULT)
+
+## X and Y dimensions
+MATRIX_X := 110
+MATRIX_Y := 9
+
+## pixel size
+SDL_SCALE_FACTOR := 1
+
+# use the xtensa toolchain
+CFLAGS += -mlongcalls
+CFLAGS += -ffunction-sections
+CFLAGS += -fdata-sections
+CFLAGS += -fstrict-volatile-bitfields
+# TODO add dependencies to ext:
+CFLAGS += -I/home/cyrill/.platformio/packages/framework-arduinoespressif32/tools/sdk/include/freertos/
+CFLAGS += -I/home/cyrill/.platformio/packages/framework-arduinoespressif32/tools/sdk/include/config/
+CFLAGS += -I/home/cyrill/.platformio/packages/framework-arduinoespressif32/tools/sdk/include/esp32/
+CFLAGS += -I/home/cyrill/.platformio/packages/framework-arduinoespressif32/tools/sdk/include/soc/
+CFLAGS += -I/home/cyrill/.platformio/packages/framework-arduinoespressif32/tools/sdk/include/heap/
+CFLAGS += -I/home/cyrill/.platformio/packages/framework-arduinoespressif32/tools/sdk/include/driver/
+CFLAGS += -I/home/cyrill/.platformio/packages/framework-arduinoespressif32/tools/sdk/include/driver/
+CFLAGS += -I/home/cyrill/.platformio/packages/framework-arduinoespressif32/tools/sdk/include/esp_ringbuf/
+
+# Use the xtensa toolchain
+CC := xtensa-esp32-elf-cc
```

Comment:
I installed it via package manager. But probably platformio ships those too (shouldn't matter I think).

Reply:
Well, it's probably a good idea to use the same toolchain used to build the other stuff.
Luckily it's not so much of a problem as it might've been with C++ code.
Probably a good idea to default to platformio's toolchain for consistency.

Reply:
ACK. I have the toolchain available also in the ~/.platformio path so I assume it ships with it. We could do something like that?
```suggestion
CC := ${PLATFORMIO_HOME}/packages/toolchain-xtensa32/bin/xtensa-esp32-elf-cc
AR := ${PLATFORMIO_HOME}/packages/toolchain-xtensa32/bin/xtensa-esp32-elf-ar
```

Reply:
Yeah, sounds good.